daemon/logger/gelf/gelf.go

```diff
@@ -251,23 +250,18 @@ func parseAddress(address string) (*url.URL, error) {
 	if address == "" {
 		return nil, fmt.Errorf("gelf-address is a required parameter")
 	}
-	if !urlutil.IsTransportURL(address) {
-		return nil, fmt.Errorf("gelf-address should be in form proto://address, got %v", address)
-	}
-	url, err := url.Parse(address)
+	addr, err := url.Parse(address)
```

Comment:
Hmm, are there cases where `url.Parse` might infer or assign a scheme that we didn't intend?  The check in `urlutil.IsTransportURL` is prefix-based, so should be pretty performant (and feels safer than trusting `url.Parse`, IMO).

I guess I'd be more worried about it inferring `file://` than `udp://` or `tcp://`, so this is probably fine. :see_no_evil:

Reply:
I think we should be fine; couple of lines down we check if the scheme is supported;

```go
if addr.Scheme != "udp" && addr.Scheme != "tcp" {
```

I can't come up with cases where it could invalidate a _valid_ address (at most one that's specified without scheme, but that should produce an error message, helping the user to find the problem)